src/mpi/comm/commutil.c

```diff
@@ -587,6 +587,14 @@ int MPIR_Comm_create_subcomms(MPIR_Comm * comm)
         comm->node_comm->local_size = num_local;
         comm->node_comm->remote_size = num_local;
 
+        /* Copy relevant hints to node_comm */
+        comm->node_comm->hints[MPIR_COMM_HINT_VCI_IDX_SENDER] =
+            comm->hints[MPIR_COMM_HINT_VCI_IDX_SENDER];
+        comm->node_comm->hints[MPIR_COMM_HINT_VCI_IDX_RECEIVER] =
+            comm->hints[MPIR_COMM_HINT_VCI_IDX_RECEIVER];
+        comm->node_comm->hints[MPIR_COMM_HINT_THREAD_ID] = comm->hints[MPIR_COMM_HINT_THREAD_ID];
```

Comment:
They are registered inside ch4, but leaked here in the MPIR layer. Is it possible to do this in the ch4-layer?

Reply:
That is right in on way.  However, with the current design, these hints are defined at mpir (in src/include/mpir_comm.h) but registered at ch4. So, it seemed keeping here may be okay. Do you still favor moving it?

Reply:
Alright, let's keep it simple (as is) for now.